### ISSUE TITLE
Can't find flutter sdk when opening specific folder on ubuntu.

### ISSUE THREAD

**Issue by bsutton** [Dart-Code/Dart-Code#1922]
I have multiple flutter project on my ubuntu 19.04 PC.

I have an application type flutter project and a third party plugin I've pulled down from github.

Note: these test were conducted after rebooting my PC to ensure all environment changes had taken affect.
My path is: 
```
/home/bsutton/.cargo/bin:/opt/gradle/gradle-5.2.1/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/bsutton/apps/flutter/bin/cache/dart-sdk/bin:/home/bsutton/apps/flutter/bin
```

The flutter logs however show:

```
[10:45:44 am] [General] [Info] Environment PATH:
[10:45:44 am] [General] [Info]     /home/bsutton/.cargo/bin
[10:45:44 am] [General] [Info]     /opt/gradle/gradle-5.2.1/bin
[10:45:44 am] [General] [Info]     /usr/local/sbin
[10:45:44 am] [General] [Info]     /usr/local/bin
[10:45:44 am] [General] [Info]     /usr/sbin
[10:45:44 am] [General] [Info]     /usr/bin
[10:45:44 am] [General] [Info]     /sbin
[10:45:44 am] [General] [Info]     /bin
[10:45:44 am] [General] [Info]     /usr/games
[10:45:44 am] [General] [Info]     /usr/local/games
[10:45:44 am] [General] [Info]     /snap/bin
```

Note that this list is identical for both flutter projects however the my_app project still finds flutter.

When I open my application (my_app) project vscode finds the flutter sdk using my path as you can see in the below logs:
```
Looking for flutter in:
[10:45:44 am] [General] [Info]         /home/bsutton/git/my_app
[10:45:44 am] [General] [Info]         /home/bsutton/git/my_app/bin
[10:45:44 am] [General] [Info]         /home/bsutton/apps/flutter/bin/
[10:45:44 am] [General] [Info]         /home/bsutton/.cargo/bin
[10:45:44 am] [General] [Info]         /opt/gradle/gradle-5.2.1/bin
[10:45:44 am] [General] [Info]         /usr/local/sbin
[10:45:44 am] [General] [Info]         /usr/local/bin
[10:45:44 am] [General] [Info]         /usr/sbin
[10:45:44 am] [General] [Info]         /usr/bin
[10:45:44 am] [General] [Info]         /sbin
[10:45:44 am] [General] [Info]         /bin
[10:45:44 am] [General] [Info]         /usr/games
[10:45:44 am] [General] [Info]         /usr/games/bin
[10:45:44 am] [General] [Info]         /usr/local/games
[10:45:44 am] [General] [Info]         /usr/local/games/bin
[10:45:44 am] [General] [Info]         /snap/bin
[10:45:44 am] [General] [Info]     Found at:
[10:45:44 am] [General] [Info]         /home/bsutton/apps/flutter/bin/
[10:45:44 am] [General] [Info]     Candidate paths to be post-filtered:
[10:45:44 am] [General] [Info]         /home/bsutton/apps/flutter
[10:45:44 am] [General] [Info]     Found at /home/bsutton/apps/flutter
[10:45:44 am] [General] [Info]     Returning SDK path /home/bsutton/apps/flutter for flutter
```
When I start vscode (via New Window or launching a new instance) with the plugin project the logs show:

```
[10:44:57 am] [General] [Info] Searching for flutter
[10:44:57 am] [General] [Info]     Looking for flutter in:
[10:44:57 am] [General] [Info]         /home/bsutton/git/flutter_sound/flutter_sound
[10:44:57 am] [General] [Info]         /home/bsutton/git/flutter_sound/flutter_sound/bin
[10:44:57 am] [General] [Info]         /home/bsutton/.cargo/bin
[10:44:57 am] [General] [Info]         /opt/gradle/gradle-5.2.1/bin
[10:44:57 am] [General] [Info]         /usr/local/sbin
[10:44:57 am] [General] [Info]         /usr/local/bin
[10:44:57 am] [General] [Info]         /usr/sbin
[10:44:57 am] [General] [Info]         /usr/bin
[10:44:57 am] [General] [Info]         /sbin
[10:44:57 am] [General] [Info]         /bin
[10:44:57 am] [General] [Info]         /usr/games
[10:44:57 am] [General] [Info]         /usr/games/bin
[10:44:57 am] [General] [Info]         /usr/local/games
[10:44:57 am] [General] [Info]         /usr/local/games/bin
[10:44:57 am] [General] [Info]         /snap/bin
[10:44:57 am] [General] [Info]     Found at:
[10:44:57 am] [General] [Info]     Candidate paths to be post-filtered:
[10:44:57 am] [General] [Info]     Returning SDK path undefined for flutter
```
You can see from the above logs that the second project stops searching my path just before the flutter directory.
This makes no sense.

I've also tried setting the directory via the button that pops up 'Locate SDK' and vscode just ignores the directory I set.

**Comment by bsutton** [comment]
OK, so found at least part of the problem.

I had set my PATH using .bashrc and .bash_profile. Of course these only set the PATH for a bash shell.
As such if I launched vscode from anywhere other than bash it wouldn't see the path.

After moving my path settings to ~/.profile and rebooted vscode correctly found my path.

So the path issue was my stupidity however it still leaves the questions as to why after select the flutter sdk path manually via the  'Locate SDK' button vscode still couldn't find the  sdk?

**Comment by DanTup** [comment]
> I had set my PATH using .bashrc and .bash_profile. Of course these only set the PATH for a bash shell.
> As such if I launched vscode from anywhere other than bash it wouldn't see the path.

Yeah, this is a common issue :( We can only search in the `PATH` that the VS Code process sees.

>  it still leaves the questions as to why after select the flutter sdk path manually via the 'Locate SDK' button vscode still couldn't find the sdk?

This is indeed a bug and there's a fix in the next release (#1896). Sorry for the confusion and inconvenience!
